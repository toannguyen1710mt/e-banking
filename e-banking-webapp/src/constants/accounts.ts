// Interfaces
import { AccountType } from '@/interfaces';

export const ACCOUNT_TYPES = {
  MAIN_WALLET: 'Main Wallet',
  SAVINGS: 'Savings',
  CHECKING: 'Checking',
};

export const TRANSFER_FORM_ACCOUNT_OPTIONS = [
  {
    key: AccountType.MAIN,
    label: 'Main Account',
  },
  {
    key: AccountType.SAVINGS,
    label: 'Savings',
  },
  {
    key: AccountType.CHECKING,
    label: 'Checking',
  },
];
