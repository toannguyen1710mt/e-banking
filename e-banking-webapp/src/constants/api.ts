export const API_ENDPOINTS = {
  USERS: '/users',
  ACCOUNTS: '/accounts',
  GLOBAL_ACCOUNTS: '/global-accounts',
  TRANSACTIONS: '/transactions',
  CARDS: '/cards',
  SIGN_UP: '/auth/local/register',
  CHANGE_PASSWORD: '/auth/change-password',
};

export const BASE_URL = process.env.NEXT_PUBLIC_BASE_URL;

export const END_POINT = {
  UPLOAD: `/upload`,
};
