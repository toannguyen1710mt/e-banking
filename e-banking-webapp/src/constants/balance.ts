import { CardStatistics } from '@/interfaces';

export const EXPENSE_LABEL_DATA = ['Amazon', 'Netflix', 'Apple', 'Figma'];
export const EXPENSE_VALUE_DATA = ['#8AB17D', '#2A9D8F', '#F4A261', '#E9C46A'];
export const EXPENSE_COLOR_CLASSES = [
  'bg-[#8AB17D]',
  'bg-[#2A9D8F]',
  'bg-[#F4A261]',
  'bg-[#E9C46A]',
];
export const AVAILABLE_WALLETS = [
  CardStatistics.Savings,
  CardStatistics.Checking,
];
export const CARD_STATISTICS = [
  {
    label: CardStatistics.Balance,
    color: 'bg-java',
  },
  {
    label: CardStatistics.Savings,
    color: 'bg-pastelYellow',
  },
  {
    label: CardStatistics.Checking,
    color: 'bg-softGreen',
  },
];
