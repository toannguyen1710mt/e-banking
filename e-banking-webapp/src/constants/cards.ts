export const MODAL_STYLES_ADD_CREDIT_CARD = {
  true: {
    cardMaxWidth: 'max-w-[195px]',
    cardHeaderGap: 'gap-0',
    headerContentGap: 'gap-1',
    payPassIcon: 'w-[11px] h-[14px]',
    cardBody: 'mb-[10px] mt-[8px]',
    chipSize: 'w-[24px] h-[16px]',
    worldTextSize: 'text-5xs',
    cardNumberText: 'ml-2 mt-0 text-xs',
    detailsGap: 'mt-0 gap-2',
    expireText: 'text-3xs',
    holderNameText: 'max-w-[115px] text-3xs',
    mastercardIconSize: 'w-[35px]',
  },
  false: {
    cardMaxWidth: 'max-w-[290px]',
    cardHeaderGap: 'gap-2',
    headerContentGap: 'gap-3',
    payPassIcon: 'unset',
    cardBody: 'mb-[18px] mt-[14px]',
    chipSize: 'unset',
    worldTextSize: 'text-2xs',
    cardNumberText: 'ml-4 mt-1 text-sm',
    detailsGap: 'mt-2 gap-5',
    expireText: 'text-2xs',
    holderNameText: 'max-w-[185px] text-xs',
    mastercardIconSize: 'unset',
  },
};
