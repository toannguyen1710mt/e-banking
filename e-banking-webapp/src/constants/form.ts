// Interfaces
import { AccountType, CurrencyUnit } from '@/interfaces';

export const SIGNUP_FORM_DEFAULT_VALUES = {
  user: {
    username: '',
    email: '',
    password: '',
    confirmPassword: '',
  },
  contact: {
    phone: '',
    country: 'United States',
    postal: '344',
  },
  card: {
    holderName: '',
    cardNumber: '',
    expireAt: '',
    ccv: '',
  },
};

export const ACCOUNT_DEFAULT_VALUES = [
  {
    accountNumber: '123456789208',
    balance: 3438800,
    type: AccountType.MAIN,
    currency: '$' as CurrencyUnit,
    name: 'OCB',
  },
  {
    accountNumber: '999332289888',
    balance: 84438800,
    type: AccountType.CHECKING,
    currency: '$' as CurrencyUnit,
    name: 'ACB',
  },
  {
    accountNumber: '106868929357',
    balance: 9238800,
    type: AccountType.SAVINGS,
    currency: '$' as CurrencyUnit,
    name: 'VID',
  },
];

export const PASSWORD_DEFAULT_VALUES = {
  currentPassword: '',
  newPassword: '',
  confirmPassword: '',
};
