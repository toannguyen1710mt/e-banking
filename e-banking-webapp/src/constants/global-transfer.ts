import { GlobalType } from '@/interfaces';

export const GLOBAL_TYPES = {
  SOUTH_AFRICA: 'South Africa',
  KENYA: 'Kenya',
  GERMANY: 'Germany',
  UK: 'UK',
};

export const TRANSFER_FORM_GLOBAL_OPTIONS = [
  {
    key: GlobalType.SOUTH_AFRICA,
    label: 'South Africa',
  },
  {
    key: GlobalType.KENYA,
    label: 'Kenya',
  },
  {
    key: GlobalType.GERMANY,
    label: 'Germany',
  },
  {
    key: GlobalType.UK,
    label: 'UK',
  },
];

export const OPTIONS_COUNTRY_CODE_CONVERT_GLOBAL = [
  {
    key: GlobalType.SOUTH_AFRICA,
    label: 'ZAR',
  },
  {
    key: GlobalType.KENYA,
    label: 'KSH',
  },
  {
    key: GlobalType.GERMANY,
    label: 'EUR',
  },
  {
    key: GlobalType.UK,
    label: 'GBP',
  },
];

export const EXCHANGE_RATES_TO_USD = {
  [GlobalType.SOUTH_AFRICA]: 18.08,
  [GlobalType.KENYA]: 100,
  [GlobalType.GERMANY]: 0.95,
  [GlobalType.UK]: 0.79,
};
