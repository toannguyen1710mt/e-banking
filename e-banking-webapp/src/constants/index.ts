export * from './accounts';
export * from './routes';
export * from './themes';
export * from './messages';
export * from './images';
export * from './regex';
export * from './navbar';
export * from './titles';
export * from './global-transfer';
export * from './balance';
export * from './charts';
export * from './form';
export * from './api';
export * from './tags';
export * from './settings';
export * from './tables';
export * from './cards';
export * from './date';
export * from './sidebar-list';
