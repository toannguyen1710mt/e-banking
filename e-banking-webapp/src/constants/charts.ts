import { ApexOptions } from 'apexcharts';
import { EXPENSE_LABEL_DATA, EXPENSE_VALUE_DATA } from './balance';

export const createExpenseAnalysisOptions = (value: string): ApexOptions => ({
  chart: {
    type: 'donut',
  },
  labels: EXPENSE_LABEL_DATA,
  responsive: [
    {
      breakpoint: 320,
      options: {
        chart: {
          width: '100%',
        },
        legend: {
          position: 'bottom',
        },
      },
    },
  ],
  legend: {
    show: false,
  },
  dataLabels: {
    enabled: false,
  },
  plotOptions: {
    pie: {
      donut: {
        size: '70%',
        labels: {
          show: true,
          name: {
            show: false,
          },
          value: {
            show: true,
            fontSize: '8px',
            fontWeight: 800,
            color: '#fff',
            offsetY: 3,
          },
          total: {
            show: true,
            formatter: () => {
              const formattedValue = `${value}`;

              return formattedValue.length > 7
                ? `${formattedValue.slice(0, 5)}...`
                : formattedValue;
            },
          },
        },
      },
    },
  },
  stroke: {
    width: 1,
  },

  colors: EXPENSE_VALUE_DATA,
});
