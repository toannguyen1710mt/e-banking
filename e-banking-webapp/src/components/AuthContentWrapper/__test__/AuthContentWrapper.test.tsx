import { render } from '@testing-library/react';
import { usePathname } from 'next/navigation';

// Constants
import { ROUTES } from '@/constants';

// Components
import { AuthContentWrapper } from '..';

jest.mock('next/navigation', () => ({
  usePathname: jest.fn(),
}));

describe('AuthContentWrapper component', () => {
  const mockUsePathname = usePathname as jest.Mock;

  it('Should render snapshot correctly.', () => {
    const props = {
      formContent: <>Form</>,
      textHeading: 'Good To See You Again!',
      textFooter: 'Don’t have an account?',
    };

    expect(render(<AuthContentWrapper {...props} />)).toMatchSnapshot();
  });

  it('Should display the correct heading and footer text.', () => {
    mockUsePathname.mockReturnValue(ROUTES.SIGN_IN);
    const props = {
      formContent: <>Form</>,
      textHeading: 'Welcome Back!',
      textFooter: 'Don’t have an account?',
    };

    const { getByRole, getByText } = render(<AuthContentWrapper {...props} />);

    expect(getByRole('heading', { level: 1 })).toHaveTextContent(
      'Welcome Back!',
    );
    expect(getByText('Don’t have an account?')).toBeInTheDocument();
  });
});
