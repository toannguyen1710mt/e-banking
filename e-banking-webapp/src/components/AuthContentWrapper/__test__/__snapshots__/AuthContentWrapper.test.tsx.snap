// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthContentWrapper component Should render snapshot correctly. 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mx-auto flex flex-1 flex-col"
      >
        <img
          alt="Logo EBanking"
          class="pb-[14px] lg:pb-9"
          data-nimg="1"
          decoding="async"
          height="36"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=256&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=128&q=75 1x, /_next/image?url=%2Fimg.jpg&w=256&q=75 2x"
          style="color: transparent;"
          width="104"
        />
        <div
          class="max-w-[406px] text-center md:text-left"
        >
          <h1
            class="text-foreground-100 text-4xl font-semibold mb-0 md:mb-7"
          >
            Good To See You Again!
          </h1>
          <p
            class="text-foreground-300 text-base mb-4 font-normal leading-[22px]"
          />
        </div>
        <div
          class="mx-auto mb-[17px] max-w-[325px] md:mx-0"
        >
          Form
        </div>
        <p
          class="text-foreground-100 text-sm font-medium text-center leading-[20px] md:text-left"
        >
          Don’t have an account?
           
          <a
            class="leading-[20px] text-primary-200"
            href="/sign-in"
          >
            Sign In
          </a>
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mx-auto flex flex-1 flex-col"
    >
      <img
        alt="Logo EBanking"
        class="pb-[14px] lg:pb-9"
        data-nimg="1"
        decoding="async"
        height="36"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=256&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=128&q=75 1x, /_next/image?url=%2Fimg.jpg&w=256&q=75 2x"
        style="color: transparent;"
        width="104"
      />
      <div
        class="max-w-[406px] text-center md:text-left"
      >
        <h1
          class="text-foreground-100 text-4xl font-semibold mb-0 md:mb-7"
        >
          Good To See You Again!
        </h1>
        <p
          class="text-foreground-300 text-base mb-4 font-normal leading-[22px]"
        />
      </div>
      <div
        class="mx-auto mb-[17px] max-w-[325px] md:mx-0"
      >
        Form
      </div>
      <p
        class="text-foreground-100 text-sm font-medium text-center leading-[20px] md:text-left"
      >
        Don’t have an account?
         
        <a
          class="leading-[20px] text-primary-200"
          href="/sign-in"
        >
          Sign In
        </a>
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
